pkg/workflow/tasks/custom/task.go

```diff
@@ -290,7 +294,14 @@ func buildValueForStatus(ctx wfContext.Context, step v1beta1.WorkflowStep, pd *p
 	statusTemplate += fmt.Sprintf("status: %s\n", status)
 	statusTemplate += contextTempl
 	statusTemplate += "\n" + inputsTempl
-	return value.NewValue(template+"\n"+statusTemplate, pd, "")
+	v, err := value.NewValue(template+"\n"+statusTemplate, pd, "")
+	if err != nil {
+		return nil, err
+	}
+	if v.Error() != nil {
+		return nil, v.Error()
+	}
```

Comment:
will it bring incompatible on this error check?

Reply:
No, we want no error when validating the `if`.